artemis-core-client/src/main/java/org/apache/activemq/artemis/api/core/JsonUtil.java

```diff
@@ -275,11 +274,11 @@ public static JsonArrayBuilder toJsonArrayBuilder(byte[] byteArray) {
    }
 
    public static JsonArray readJsonArray(String jsonString) {
-      return Json.createReader(new StringReader(jsonString)).readArray();
```

Comment:
I tried calling listConsumers on tests and I couldn't reproduce it.

I'm not saying I don't believe you.. I just don't know what's being fixed without a test beyond the documentaiton you described on the PR comment.

Can you provide a test?

Reply:
Hi, clebertsuconic, we dived into code today and checked our deployed production configuration. Found that we didn't use the etc/management.xml and instead added jmx configuration in artemis.profile  like "-Dcom.sun.management.jmxremote=true -Dcom.sun.management.jmxremote.port=1099". In such a way, the rmi thread pool exeuctor is using system classloader. When requests from jconsole come in, the exception occcurs.

The reason ActiveMQServerControlTest can't reproduce the problem is the test uses junit class loader which contains all the necessary url paths.

Reply:
With regard to test, we tried and it was somewhat tricky(we had to add jmx configurations in system property and executed "artemis run" in a standalone process).

By the way, previous artemis server exported the jmx this way. Although we upgraded our server, we kept the original configs(including jmx configs) to make the operations more efficient and less error-prone.

Reply:
@wy96f the soak-tests will use a real server for testing. There's even one for JMX testing. woudn't that be a good suitable place?

Reply:
I managed to reproduce this issue once by the way @wy96f described seems a valid bug if not using jololkia but native jmx